GoMeet/src/main/webapp/js/vote-meeting-time.js

```diff
@@ -0,0 +1,178 @@
+      if (time.voters !== null && time.voters.includes(currentUser)) {
+        votedTimes.add(time.id);
+      }
+      timeData.push(time);
+    }
+  }
+  generateVoteTimeForm(timeData, currentUser, votedTimes);
+}
+
+/**
+ * Generates a voting form containing the MeetingTime timeData 
+ * in the 'vote-meeting-times' <div>. 
+ * Current logged-in user will use this form to vote on the 
+ * associated meeting times.
+ * @param {Array[Object]} timeData An array of timeData objects
+ * retrieved from the '/meeting-time' servlet via GET request, in 
+ * the format:
+ * {meetingTimeId: id, datetime: datetimeStr, voteCount: count, voters: [voters]}
+ */
+function generateVoteTimeForm(timeData, currentUser, votedTimes) {
+  let votingRight = true;
+  if (votedTimes.size === MAX_VOTES) {
+    votingRight = false; // User may not vote again if they have voted maximum times.
+  }
+  // Disable all vote buttons for users that have voted MAX_VOTES already.
+  let disabledState = !votingRight; 
+
+  sortTimesbyVotes(timeData);
+
+  let voteTable = document.getElementById('meeting-times-table');
+  
+  for (let i = 0; i < timeData.length; i++) {
+    let time = timeData[i];
+    let timeRow = voteTable.insertRow(i + 1);
+
+    let dateTimeCell = timeRow.insertCell(0);
+    dateTimeCell.innerText = time.datetime;
+
+    let voteCountCell = timeRow.insertCell(1);
+    voteCountCell.innerText = time.voteCount;
+
+    let votersCell = timeRow.insertCell(2);
+    votersCell.innerText = time.voters;
+
+    // Clicking on this button will cast a vote for this meeting time.
+    let voteFormCell = timeRow.insertCell(3);
+    let voteButton = document.createElement('button');
+    voteButton.innerText = 'Vote Time'
+    voteButton.id = time.id;
+    voteButton.onclick = function() {voteTime(time.id, currentUser)};
+    if (votedTimes.has(time.id)) {
+      // Always disable buttons of times the user has voted for already.
+      voteButton.disabled = true;
+    } else {
+      voteButton.disabled = disabledState;
+    }
+    voteFormCell.appendChild(voteButton);
+  }
+}
+
+/**
+ * Sorts the timeData objects given by their 'voteCount'
+ * property in descending order.
+ * NOTE: Sorting the timeData objects client-side, because likely not every single 
+ * client-side use case of the timeData requires the data to be sorted in vote order. 
+ * Hence, sorting the timeData server-side will add unnecessary additional complexity 
+ * to the server code (and consequently the fetch() calls to the servlet).
+ * @param {Array[Object]} timeData an array of timeData objects
+ * retrieved from the '/meeting-time' servlet via GET request, in 
+ * the format:
+ * {meetingTimeId: id, datetime: datetimeStr, voteCount: count, voters: [voters]}
+ */
+function sortTimesbyVotes(timeData) {
+  // Sort the times by votes, in decreasing order.
+  timeData.sort((time1, time2) => {
+    if (time1.voteCount > time2.voteCount) {
```

Comment:
Seems like you could have done something like `return time2.voteCount - time1.voteCount` instead, is there a reason you did it this way?

Reply:
No, I just thought that the number returned had to be -1, 0, or 1 😅 But a closer look at the docs says it's just either negative, 0 or positive, so I've refactored it.